.dotnet/README.md

```diff
@@ -795,6 +796,52 @@ string message = outputAsJson.RootElement
 
 Notice how you can then call the resulting `ClientResult`'s `GetRawResponse` method and retrieve the response body as `BinaryData` via the `PipelineResponse`'s `Content` property.
 
+### Mock a client for testing
+
+The OpenAI .NET library has been designed to support mocking, providing key features such as:
+- Client methods made virtual to allow overriding.
+- Model Factories to assist in instantiating models that lack public constructors.
```

Comment:
nit: 
1. Do not capitalize "Model Factory"
2. I wonder if calling them "API models" helps disambiguate between these models and LLMs such as "gpt-4o".

In other words:
```text
- Model factories to assist in instantiating API output models that lack public constructors.
```